llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp

```diff
@@ -4634,7 +4633,7 @@ bool ARMAsmParser::parseRegisterList(OperandVector &Operands, bool EnforceOrder,
     Reg = getDRegFromQReg(Reg);
     EReg = MRI->getEncodingValue(Reg);
     Registers.emplace_back(EReg, Reg);
-    ++Reg;
+    Reg = Reg + 1;
```

Comment:
Should we add a operator++ to MCRegister?

Reply:
IIRC registers are emitted in the order they are defined in td, which makes ++ work most of the time, but there is no guarantee in general.

In this particular case it should probably be `Reg = MRI->getSubReg(Reg, ARM::dsub_1)` (see `getDRegFromQReg`).

Reply:
> IIRC registers are emitted in the order they are defined in td

No, it seems it's not that simple. They're not sorted alphabetically either.